crates/bevy_utils/src/lib.rs

```diff
@@ -233,3 +234,26 @@ impl<K: Hash + Eq + PartialEq + Clone, V> PreHashMapExt<K, V> for PreHashMap<K,
         }
     }
 }
+
+/// A type which calls a function when dropped.
+/// This can be used to ensure that cleanup code is run even in case of a panic.
+pub struct OnDrop<F: FnOnce()> {
```

Comment:
Btw, `bevy_tasks` also has a `CallOnDrop` impl which could replaced with this. Not sure if it's worth taking on `bevy_utils` as a dependency.